src/formatDate.js

```diff
@@ -9,6 +43,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const possibleDateParts = ['YYYY', 'YY', 'MM', 'DD'];
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
```

Comment:
The variable `oldSeparator` is derived from the last element of the `fromFormat` array, which could be error-prone if the array structure changes. It would be safer to explicitly specify the separator index when accessing it.